main.go

```diff
@@ -92,75 +65,54 @@ func exportOutputs(outputs map[string]string) error {
 	return nil
 }
 
-func logFail(format string, v ...interface{}) {
+func failf(format string, v ...interface{}) {
 	log.Errorf(format, v...)
 	os.Exit(1)
 }
 
 func main() {
-	//
-	// validate and prepare inputs
-	configs := createConfigsModelFromEnvs()
-
-	fmt.Println()
-	configs.print()
-
-	if err := configs.validate(); err != nil {
-		logFail("Issue with input: %s", err)
-	}
-
-	var newVersionCode int
-	if configs.NewVersionCode != "" {
-		var err error
-		newVersionCode, err = strconv.Atoi(configs.NewVersionCode)
-		if err != nil {
-			logFail("Failed to convert to string: %s, error: %s", configs.NewVersionCode, err)
-		}
-	}
-
-	if configs.VersionCodeOffset != "" {
-		offset, err := strconv.Atoi(configs.VersionCodeOffset)
-		if err != nil {
-			logFail("Failed to convert to string: %s, error: %s", configs.VersionCodeOffset, err)
-		}
-		newVersionCode += offset
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Issue with input: %s", err)
 	}
+	stepconf.Print(cfg)
+	fmt.Println()
```

Comment:
Previously there was a check, which is now missing: errors.New("neither NewVersionCode nor NewVersionName are provided however, one of them is required")